packages/amplify-category-geo/src/commands/geo/add.ts

```diff
@@ -1,16 +1,21 @@
-import { chooseServiceMessageAdd, previewBanner, provider } from '../../service-utils/constants';
-import { category } from '../../constants';
+import { chooseServiceMessageAdd, provider } from '../../service-utils/constants';
+import { category, supportedRegions } from '../../constants';
 import { supportedServices } from '../../supportedServices';
-import { $TSAny, $TSContext } from 'amplify-cli-core';
-import { addResource } from '../../provider-controllers';
+import { $TSAny, $TSContext, stateManager } from 'amplify-cli-core';
+import { addResource, unsupportedRegionError } from '../../provider-controllers';
 import { printer } from 'amplify-prompts';
 
 export const name = 'add';
 
 export const run = async(context: $TSContext) => {
   const { amplify } = context;
-
   try {
+    const region = stateManager.getMeta()?.providers[provider]?.Region;
+    if(!supportedRegions.includes(region)) {
+      printer.error(unsupportedRegionError(region));
+      return;
+    }
```

Comment:
put this logic in a function like `verifySupportedRegion` and use it across all of these entry points to the category rather than duplicating it

Reply:
Running into some issues abstracting this out. I will make sure to make this better post GA.

Reply:
What's the issue? Seems like it should be straightforward to put it in a shared utilities file, but could be missing something

Reply:
The `verifySupportedRegion` which I tried to define in `provider-controllers/index` is not being hit at all and gives back `undefined`! So weird.

Reply:
Fixed this